pkg/scheduler/objects/application.go

```diff
@@ -103,6 +104,7 @@ type Application struct {
 	stateLog             []*StateLogEntry            // state log for this application
 	placeholderData      map[string]*PlaceholderData // track placeholder and gang related info
 	askMaxPriority       int32                       // highest priority value of outstanding asks
+	hasPlaceholderAlloc  atomic.Bool                 // Whether there is at least one allocated placeholder (has to be atomic.Bool due to queue/app lock ordering)
```

Comment:
Does not need to be atomic, application sorting uses the calls to get other data from the application which require locking the application. Can use the same mechanism. Sticking with that makes sure that if a release is in progress it works.

Reply:
This is only possible if we call `GetCopyOfApps()` then go through it and filter out the ones without a PH allocation. I wanted to avoid this because we copy the apps unnecessarily. Otherwise, we have a locked mutex order queue-> application and the inverse application->queue can happen in a separate goroutine (eg. `AddAllocationAsk()`, `removeAsksInternal()`) at the same time.

Reply:
Ok, I modified it regardless, let's see if this is any good.

Reply:
Update: I used the new unit test to check performance, we really want to avoid `GetCopyOfApps()`. That slows things down significantly. I believe we still have a speedup because we avoid the actual sorting, but the speedup is much smaller this way.